ot/gromov/_utils.py

```diff
@@ -797,3 +798,342 @@ def update_barycenter_feature(
             inv_p = 1. / p_sum
 
     return sum(list_features) * inv_p[:, None]
+
+
+############################################################################
+# Methods related to fused unbalanced GW and unbalanced Co-Optimal Transport.
+############################################################################
+
+def div_to_product(pi, a, b, pi1=None, pi2=None, divergence="kl", mass=True, nx=None):
+    r"""Calculate the Bregman divergence between an arbitrary measure and a product measure.
+    This implementation induces cheaper cost than the direct calculation.
+    Only support for Kullback-Leibler and half-squared L2 divergences.
+
+    For half-squared L2 divergence:
+    .. math::
```

Comment:
check spacing + that equation holds on a page. I think there are rendering issues.